frontend/javascripts/admin/team/edit_team_modal_view.tsx

```diff
@@ -0,0 +1,145 @@
+import { MinusOutlined, PlusOutlined } from "@ant-design/icons";
+import { getEditableUsers, updateUser } from "admin/admin_rest_api";
+import { Modal, AutoComplete, Input, Spin } from "antd";
+import { DefaultOptionType } from "antd/lib/select";
+import * as React from "react";
+import { useEffect, useState } from "react";
+import { APITeam, APITeamMembership, APIUser } from "types/api_flow_types";
+import { filterTeamMembersOf, renderUsersForTeam } from "./team_list_view";
+
+type Props = {
+  onCancel: (...args: Array<any>) => any;
+  isOpen: boolean;
+  team: APITeam | null;
+};
+
+function EditTeamModalForm({ onCancel, isOpen, team }: Props) {
+  const [autoCompleteValue, setAutoCompleteValue] = useState("");
+  const onChange = (newValue: string) => setAutoCompleteValue(newValue);
+  const [users, setUsers] = useState<APIUser[] | null>(null);
+  const fetchUsers = async () => setTimeout(async () => setUsers(await getEditableUsers()), 5000); // TODO delete; only testing purposes
+  //const fetchUsers = async () => setUsers(await getEditableUsers());
+  useEffect(() => {
+    fetchUsers();
+  }, []);
+
+  if (team === null) return null;
+  const updateTeamMembership = async (user: APIUser, newTeams: APITeamMembership[]) => {
+    if (users === null) return;
+    const newUser = Object.assign({}, user, {
+      teams: newTeams,
+    });
+    const serverUser = await updateUser(newUser);
+    setUsers(users.map((oldUser) => (oldUser.id === serverUser.id ? serverUser : oldUser)));
+  };
+
+  const addTo = async (user: APIUser, team: APITeam | null) => {
+    if (team === null) return;
+    const newTeam: APITeamMembership = { id: team.id, name: team.name, isTeamManager: false }; //TODO never make manager?
+    updateTeamMembership(user, [...user.teams, newTeam]);
+  };
+
+  const removeFrom = async (user: APIUser, team: APITeam | null) => {
+    const newTeams = user.teams.filter((userteam) => team?.id !== userteam.id);
+    updateTeamMembership(user, newTeams);
+  };
+
+  const renderRemoveSpan = (user: APIUser) => {
+    if (user.isAdmin) {
+      return <span>Admin</span>;
+    }
+    return (
+      <span onClick={() => removeFrom(user, team)}>
+        <MinusOutlined /> Remove from {team?.name}
+      </span>
+    );
+  };
+
+  const renderTeamMember = (user: APIUser): DefaultOptionType => ({
+    //TODO was unsure whether clicking on the name should also remove team member; same for renderUserNotInTeam
```

Comment:
dont forget to delete

Reply:
I think, clicking on the name itself (and not on Add/Remove) should ideally do nothing (not even closing the dropdown). However, I could imagine that the closing of the dropdown is hard to avoid. If closing the dropdown is inevitable, the add/remove action should happen in my opinion. Otherwise, the user would think that something happened (because the dropdown closed) but actually nothing was changed. Does this make sense?

Reply:
yes, it does, thanks!

Reply:
I think this could be done by changing the AutoComplete component to a Select component https://4x.ant.design/components/select/